src/node_graph/node_graph.py

```diff
@@ -139,12 +139,18 @@ def ctx(self, value: Dict[str, Any]) -> None:
         self.graph_ctx.inputs._clear()
         self.graph_ctx.inputs._set_socket_value(value, link_limit=100000)
 
-    def generate_inputs(self) -> None:
+    def generate_inputs(self, names: Optional[List[str]] = None) -> None:
         """Generate group inputs from nodes."""
         self.inputs._clear()
+        if names is not None:
```

Comment:
There is  `_get_keys()` method to get the names of all nodes. You can use it to simplify the code. 
```python
        self.inputs._clear()
        all_names = self.nodes._get_keys()
        names = set(names or all_names)
        missing = names - all_names
        if missing:
            raise ValueError(
                f"The following named nodes do not exist: {missing}"
            )
        for name in names - set(BUILTIN_NODES):
            node = self.nodes[name]
            # skip linked sockets
            socket = node.inputs._copy(
                node=self.graph_inputs,
                    continue
                # add link from group inputs to node inputs
                self.add_link(self.inputs[new_key], node.inputs[key])
```
Same for the `outputs`.

Reply:
Hi @ahkole , I worked on your branch and implemented this, hope you don't mind.